### PR TITLE
Berlin-Adrienne Gispen-Stefan Culafic-FTWD

### PR DESCRIPTION
**Pull request by adrigispen**

### CODE REVIEW

File: starter-code/src/components/ContactsList.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    contactsCopy.splice(contactIndex, 1);
+    this.setState({
+      contacts: contactsCopy
+    });
+  };
+
+  addContact = () => {
+    let rand = Math.floor(Math.random() * this.props.contacts.length);
+    let contact = this.props.contacts[rand];
+    const contactsNew = this.state.contacts;
```
Comment (mjarraya) [inline starter-code/src/components/ContactsList.js]: This is a copy by reference. Prefer copying the values
```suggestion
    const contactsNew = [...this.state.contacts]; // or .slice()
   
```

File: starter-code/src/components/ContactsList.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    contactsNew.unshift(contact);
+    this.setState({ contacts: contactsNew });
+  };
+
+  sortByName = () => {
+    let sortedByName = this.state.contacts.sort((a, b) => {
```
Comment (mjarraya) [inline starter-code/src/components/ContactsList.js]: .sort() mutates the array in place. Protect your array against mutations ;)

File: starter-code/src/components/ContactsList.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
+import React, { Component } from "react";
+import Contact from "./Contact";
+
+class ContactsList extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      contacts: props.contacts.splice(0, 5),
+      sort: "none"
```
Comment (mjarraya) [inline starter-code/src/components/ContactsList.js]: This doesn't seem to be used.

File: starter-code/src/components/ContactsList.js (hunk 2 above)
Comment (adrigispen) [reply starter-code/src/components/ContactsList.js]: 👍

File: starter-code/src/components/ContactsList.js (hunk 3 above)
Comment (adrigispen) [reply starter-code/src/components/ContactsList.js]: yeah, I thought I might need it for the deletion but then realized that it was rerendering the whole thing so each elem always already had its proper index

**Pull request closed without merging**